File: models/file.go
package models

import "time"

// AppFile represents an application file release record
type AppFile struct {
	ID            int       `json:"id"`
	AppRecordID   int       `json:"app_record_id"`
	AppID         string    `json:"app_id,omitempty"`
	AppName       string    `json:"app_name,omitempty"`
	Version       string    `json:"version"`
	FileName      string    `json:"file_name"`
	DownloadURL   string    `json:"download_url"`
	FileSize      int64     `json:"file_size"`
	Description   string    `json:"description"`
	DownloadCount int       `json:"download_count"`
	IsActive      bool      `json:"is_active"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// FileDownloadLog represents a log entry for file download tracking
type FileDownloadLog struct {
	ID         int       `json:"id"`
	FileID     int       `json:"file_id"`
	AppID      string    `json:"app_id,omitempty"`
	AppName    string    `json:"app_name,omitempty"`
	FileName   string    `json:"file_name,omitempty"`
	Version    string    `json:"version,omitempty"`
	IP         string    `json:"ip"`
	IPLocation string    `json:"ip_location"`
	UserAgent  string    `json:"user_agent"`
	Referer    string    `json:"referer"`
	Channel    string    `json:"channel"`
	CreatedAt  time.Time `json:"created_at"`
}

File: web/art-design-pro/src/api/file.ts
import request from '@/utils/http'

export interface AppFileItem {
  id: number
  app_record_id: number
  app_id?: string
  app_name?: string
  version: string
  file_name: string
  download_url: string
  file_size: number
  description: string
  download_count: number
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface FileDownloadLogItem {
  id: number
  file_id: number
  app_id?: string
  app_name?: string
  file_name?: string
  version?: string
  ip: string
  ip_location: string
  user_agent: string
  referer: string
  channel: string
  created_at: string
}

// Fetch file list
export function fetchGetFiles(params?: {
  app_record_id?: number
  current?: number
  size?: number
}) {
  return request.get<{ list: AppFileItem[]; total: number }>({
    url: '/admin/files',
    params
  })
}

// Create new file release
export function fetchCreateFile(data: Partial<AppFileItem>) {
  return request.post<AppFileItem>({
    url: '/admin/files/create',
    data
  })
}

// Update file release
export function fetchUpdateFile(data: Partial<AppFileItem>) {
  return request.post<AppFileItem>({
    url: '/admin/files/update',
    data
  })
}

// Delete file release
export function fetchDeleteFile(data: { id: number }) {
  return request.post({
    url: '/admin/files/delete',
    data
  })
}

// Get download logs
export function fetchGetDownloadLogs(params?: {
  file_id?: number
  current?: number
  size?: number
}) {
  return request.get<{ list: FileDownloadLogItem[]; total: number }>({
    url: '/admin/files/logs',
    params
  })
}

// Upload local file to runtimes/uploads
export function fetchUploadFile(data: FormData, options?: { timeout?: number }) {
  return request.post<{ file_name: string; file_size: number; download_url: string }>({
    url: '/admin/files/upload',
    data,
    timeout: options?.timeout ?? 60000,
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}
